src/dmi.rs

```diff
@@ -0,0 +1,24 @@
+use std::fs::File;
+use png::{Decoder, Encoder, HasParameters};
+
+use error::Result;
+
+byond_fn! { strip_dmi_metadata(path) {
+    strip_metadata(path).err()
+} }
+
+fn strip_metadata(path: &str) -> Result<()> {
+    // Read in the PNG, discarding metadata
+    let (info, mut reader) = Decoder::new(File::open(path)?).read_info()?;
+    let mut buffer = vec![0; reader.output_buffer_size()];
+    reader.next_frame(&mut buffer)?;
+    drop(reader);
```

Comment:
Is there really any reason to explicitly call `mem::drop` here?

Reply:
We need to drop the `File` that was opened for reading so that we can open it for writing below.